src/FlaUI.Core/Logging/AbstractBaseLogger.cs

```diff
@@ -0,0 +1,233 @@
+﻿using System;
+
+namespace FlaUI.Core.Logging
+{
+    public abstract class AbstractBaseLogger : ILogger
+    {
+        public bool IsTraceEnabled { get; set;}
+        public bool IsDebugEnabled { get; set; }
+        public bool IsInfoEnabled { get; set; }
+        public bool IsWarnEnabled { get; set; }
+        public bool IsErrorEnabled { get; set; }
+        public bool IsFatalEnabled { get; set; }
+
+        public abstract void GatedTrace(string message);
```

Comment:
Thought about that, decided it was better to leave it public as one may want to throw a decorator pattern over these. Not to mention having it public makes it easier to test.